chia/data_layer/data_store.py

```diff
@@ -233,23 +233,33 @@ async def insert_into_data_store_from_file(
                 if node_hash not in internal_nodes and node_hash not in terminal_nodes:
                     missing_hashes.append(node_hash)
 
-        async with self.db_wrapper.reader() as reader:
-            for node_hash in missing_hashes:
-                cursor = await reader.execute(
-                    "SELECT root_hash, idx FROM nodes WHERE hash = ? AND store_id = ?",
-                    (
-                        node_hash,
-                        store_id,
-                    ),
-                )
+        batch_size = min(500, SQLITE_MAX_VARIABLE_NUMBER - 10)
+        found_hashes: set[bytes32] = set()
 
-                row = await cursor.fetchone()
-                if row is None:
-                    raise Exception(f"Unknown hash {node_hash.hex()}")
+        async with self.db_wrapper.reader() as reader:
+            for i in range(0, len(missing_hashes), batch_size):
+                batch = missing_hashes[i : i + batch_size]
```

Comment:
we have a helper function for this: https://github.com/Chia-Network/chia-blockchain/blob/main/chia/util/batches.py#L18